plugins/mac-avcapture/OBSAVCapture.m

```diff
@@ -700,21 +628,67 @@ + (NSString *)stringFromSettings:(void *)settings withSetting:(NSString *)settin
     return result;
 }
 
++ (const char *)effectsWarningForDevice:(AVCaptureDevice *)device
```

Comment:
I was wondering if this should return an `NSString *` instead, and we wait to convert it to a C string until after we call this from `plugin-properties.m`. That would allow us to remove the `<obs-module.h>` import above.

Also curious if there is a good way to check compile-time SDK availability, since the background replacement code that I've commented out here will only compile with the macOS 15 SDK.

Reply:
I was about to suggest that - try to stay within Foundation and Cocoa as much as possible and only convert to pure C types at the boundary of the `libobs` API functions.

Reply:
Changed, import removed.